cmd/ci-operator/main.go

```diff
@@ -864,6 +866,50 @@ func inputHash(inputs api.InputDefinition) string {
 	return oneWayNameEncoding.EncodeToString(hash.Sum(nil)[:5])
 }
 
+// saveNamespaceArtifacts is a best effort attempt to save ci-operator namespace artifacts to disk
+// for review later.
+func (o *options) saveNamespaceArtifacts() {
+	if len(o.artifactDir) == 0 {
+		return
+	}
+
+	namespaceDir := filepath.Join(o.artifactDir, "build-resources")
+	if err := os.Mkdir(namespaceDir, 0777); err != nil {
+		log.Printf("Unable to create build-resources directory: %v", err)
+		return
+	}
+
+	if kubeClient, err := coreclientset.NewForConfig(o.clusterConfig); err == nil {
+		pods, _ := kubeClient.Pods(o.namespace).List(meta.ListOptions{})
+		data, _ := json.MarshalIndent(pods, "", "  ")
```

Comment:
Shouldn't we technically be using the Serializer here?

Reply:
It's not required, serializer is just a thin wrapper.  We require that our objects decode correctly for other reasons